dispatch/dispatch.go

```diff
@@ -454,6 +469,34 @@ func newAggrGroup(ctx context.Context, labels model.LabelSet, r *Route, to func(
 	return ag
 }
 
+func (ag *aggrGroup) nextTick(now time.Time) (time.Duration, error) {
+	entries, err := ag.nflog.Query(
+		nflog.QGroupKey(ag.GroupKey()),
+		nflog.QReceiver(ag.logRecv),
+	)
+	if err != nil && !errors.Is(err, nflog.ErrNotFound) {
+		return 0, fmt.Errorf("error querying log entry: %w", err)
+	} else if errors.Is(err, nflog.ErrNotFound) || len(entries) == 0 {
+		level.Info(ag.logger).Log("msg", "log entry not found", "flush_time", "now", now)
+		return 0, nil
+	} else if len(entries) > 1 {
+		return 0, fmt.Errorf("unexpected entry result size: %d", len(entries))
+	}
+
+	level.Info(ag.logger).Log("msg", "found log entry", "entry", entries[0], "flush_time", entries[0].FlushTime, "now", now)
+
+	ft := entries[0].FlushTime
+	if (ft == nil) || (*ft == time.Time{}) {
+		return 0, fmt.Errorf("flush time nil or empty")
```

Comment:
I think nextTick should not return error. In case of the problems, we should fallback to the default behavior and flush anyway.